aspect/src/main/java/tech/ydb/yoj/aspect/tx/YojTransactionAspect.java

```diff
@@ -48,12 +48,19 @@ private Object doInTransaction(ProceedingJoinPoint pjp, YojTransactional transac
             if (transactional.maxRetries() != YojTransactional.UNDEFINED) {
                 localTx = tx.withMaxRetries(transactional.maxRetries());
             }
-            return transactional.readOnly() ? localTx.readOnly().run(() -> safeCall(pjp)) : localTx.tx(() -> safeCall(pjp));
+            validateIsolationLevel(transactional);
+            return transactional.readOnly() ? localTx.readOnly().withStatementIsolationLevel(transactional.isolation()).run(() -> safeCall(pjp)) : localTx.tx(() -> safeCall(pjp));
```

Comment:
please make this into an explicit `if-else`: after introducing the logic with `transactional.isolation()`, the ternary operator is now unreadable